pkg/controller/serviceaccount/tokengetter.go

```diff
@@ -43,6 +43,19 @@ func NewGetterFromClient(c clientset.Interface, secretLister v1listers.SecretLis
 	return clientGetter{c, secretLister, serviceAccountLister, podLister, nodeLister}
 }
 
+// NewNodelessGetterFromClient returns a ServiceAccountTokenGetter that
```

Comment:
I don't understand why we need this variant... the inputs are interfaces, if you want to run in a nodeless or podless environment, pass in implementations that return "unimplemented" errors

Reply:
I think you right here. Will give it one more try. Was experimenting with ways to make it less intrusive, but this does not look like right approach.